gremlin-go/driver/connection.go

```
@@ -19,50 +19,64 @@ under the License.
 
 package gremlingo
 
+import (
+	"errors"
+)
+
+type connectionState int
+
+const (
+	initialized connectionState = iota + 1
+	established
+	closed
+	closedDueToError
+)
+
 type connection struct {
-	host            string
-	port            int
-	transporterType TransporterType
-	logHandler      *logHandler
-	transporter     transporter
-	protocol        protocol
-	results         map[string]ResultSet
+	logHandler *logHandler
+	protocol   protocol
+	results    map[string]ResultSet
+	state      connectionState
 }
 
-func (connection *connection) close() (err error) {
-	if connection.transporter != nil {
-		err = connection.transporter.Close()
-	}
-	return
+func (connection *connection) errorCallback() {
+	connection.logHandler.log(Error, errorCallback)
+	connection.state = closedDueToError
+	_ = connection.protocol.close()
```

Any reason we are swallowing this error?